functions/java-jre.bash

```diff
@@ -1,26 +1,33 @@
 #!/usr/bin/env bash
 # shellcheck disable=SC2181
 
-## Install appropriate Java version based on current platform.
-## Valid arguments: 64-bit, 32-bit
+## Install appropriate Java version based on current choice.
+## Valid arguments: AdoptOpenJDK, Zulu32, Zulu64
 ##
 ##    java_install_and_update(String arch)
 ##
 java_install_or_update(){
-  # Make sure we don't overwrite existing non Java Zulu installations
-  if ! [ -x "$(command -v java)" ] || [[ ! "$(java -version)" == *"Zulu"* ]]; then
-    if [ "$1" == "64-bit" ]; then
-      if is_x86_64; then
-        java_zulu_enterprise_8_apt
-      else
-        if java_zulu_tar_update_available 64-bit; then
-          java_zulu_8_tar 64-bit
-        fi
+  # Make sure we don't overwrite existing unsupported installations
+  if ! [ -x "$(command -v java)" ] || [[ ! "$(java -version)" == *"Zulu"* ]] || [[ ! "$(java -version)" == *"AdoptOpenJDK"* ]]; then
+    if [ "$1" == "AdoptOpenJDK" ]; then
+      adoptopenjdk_install_apt
+    else
+      if dpkg -s 'adoptopenjdk-8-hotspot-jre' >/dev/null 2>&1; then
+        adoptopenjdk_uninstall_apt
```

Comment:
as commented already, I wouldn't want you do uninstall but just change where java points to
think offline...

Reply:
Fixed